test/e2e_node/pod_resize_test.go

```diff
@@ -0,0 +1,1768 @@
+				resizedPod := e2epod.WaitForPodResizeActuation(ctx, f, podClient, newPod, patchedPod, expectedContainers, initialContainers, isRollback)
+
+				// Check cgroup values only for containerd versions before 1.6.9
+				ginkgo.By(fmt.Sprintf("verifying pod container's cgroup values after %s", opStr))
+				framework.ExpectNoError(e2epod.VerifyPodContainersCgroupValues(ctx, f, resizedPod, expectedContainers))
+
+				ginkgo.By(fmt.Sprintf("verifying pod resources after %s", opStr))
+				e2epod.VerifyPodResources(resizedPod, expectedContainers)
+
+				ginkgo.By(fmt.Sprintf("verifying pod allocations after %s", opStr))
+				gomega.Eventually(ctx, e2epod.VerifyPodAllocations, timeouts.PodStartShort, timeouts.Poll).
+					WithArguments(resizedPod, expectedContainers).
+					Should(gomega.BeNil(), "failed to verify Pod allocations for resizedPod")
+				// TODO make this dynamic depending on Policy Name, Resources input and topology of target
+				// machine.
+				// For the moment skip below if CPU Manager Policy is set to none
+				if policy.name == string(cpumanager.PolicyStatic) {
+					ginkgo.By("verifying pod Cpus allowed list value after resize")
+					if isInPlacePodVerticalScalingExclusiveCPUsEnabled {
+						// TODO InPlacePodVerticalScaling alongside CPU Policy Static with InPlacePodVerticalScalingExclusiveCPUs enabled,
+						//      should pass below with tc.expected. To unblock development of fix it passes since pod resize is skipped
+						gomega.Eventually(ctx, e2epod.VerifyPodContainersCpusAllowedListValue, timeouts.PodStartShort, timeouts.Poll).
+							WithArguments(f, resizedPod, tc.containers).
+							Should(gomega.BeNil(), "failed to verify Pod CpusAllowedListValue for resizedPod with InPlacePodVerticalScalingExclusiveCPUs enabled")
+					} else {
+						gomega.Eventually(ctx, e2epod.VerifyPodContainersCpusAllowedListValue, timeouts.PodStartShort, timeouts.Poll).
+							WithArguments(f, resizedPod, tc.containers).
+							Should(gomega.BeNil(), "failed to verify Pod CpusAllowedListValue for resizedPod with InPlacePodVerticalScalingExclusiveCPUs disabled (default)")
+					}
+				}
+			}
+
+			patchAndVerify(tc.patchString, tc.expected, tc.containers, "resize", false)
+
+			rbPatchStr, err := e2epod.ResizeContainerPatch(tc.containers)
+			framework.ExpectNoError(err)
+			// Resize has been actuated, test rollback
+			patchAndVerify(rbPatchStr, tc.containers, tc.expected, "rollback", true)
+
+			ginkgo.By("deleting pod")
+			deletePodSyncByName(ctx, f, newPod.Name)
+			// we need to wait for all containers to really be gone so cpumanager reconcile loop will not rewrite the cpu_manager_state.
+			// this is in turn needed because we will have an unavoidable (in the current framework) race with the
+			// reconcile loop which will make our attempt to delete the state file and to restore the old config go haywire
+			waitForAllContainerRemoval(ctx, newPod.Name, newPod.Namespace)
```

Comment:
`deletePodSyncByName` will wait until the pod is delete. Do we need `waitForAllContainerRemoval` in that case? I would expect that the pod containers are really gone by the time pod is deleted.

Reply:
Did thorough tests, yest we need it. I was getting flaky tests.